src/Microsoft.VisualBasic/tests/VB/DriveInfo.Windows.Tests.vb

```diff
@@ -0,0 +1,174 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.VisualBasic.FileIO
+
+Imports System
+Imports System.Collections
+Imports System.Collections.Generic
+Imports System.Linq
+Imports System.Runtime.InteropServices
+Imports System.Security
+Imports Xunit
+Imports System.Text
+' DO NOT IMPORT System.IO
+Namespace Microsoft.VisualBasic.Tests
+
+    Public NotInheritable Class DriveInfoWindowsTests
+        Private Shared Sub DoDriveCheck()
+            ' Get Volume Label - valid drive
+            Dim serialNumber, maxFileNameLen, fileSystemFlags As Integer
+            Dim volNameLen As Integer = 50
+            Dim fileNameLen As Integer = 50
+            Dim volumeName As StringBuilder = New StringBuilder(volNameLen)
+            Dim fileSystemName As StringBuilder = New StringBuilder(fileNameLen)
+
+            Dim validDrive As IO.DriveInfo = FileSystem.Drives().First(Function(d) d.DriveType = IO.DriveType.Fixed)
+            Dim volumeInformationSuccess As Boolean = GetVolumeInformation(validDrive.Name, volumeName, volNameLen, serialNumber, maxFileNameLen, fileSystemFlags, fileSystemName, fileNameLen)
+
+            If volumeInformationSuccess Then
+                Assert.Equal(volumeName.ToString(), validDrive.VolumeLabel)
+            Else
+                Dim name As String = validDrive.VolumeLabel
+            End If
+        End Sub
+
+        <Fact>
+        <PlatformSpecific(TestPlatforms.Windows)>
+        Public Shared Sub TestGetDrives()
+            Dim validExpectedDrives As IEnumerable(Of Char) = GetValidDriveLettersOnMachine()
+            Dim validActualDrives() As IO.DriveInfo = FileSystem.Drives().ToArray
+
+            ' Test count
+            Assert.Equal(validExpectedDrives.Count(), validActualDrives.Count())
+
+            For i As Integer = 0 To validActualDrives.Count() - 1
+                ' Test if the drive-letter is correct
+                Assert.Contains(validActualDrives(i).Name(0), validExpectedDrives)
+            Next
+        End Sub
+
+        <Fact>
+        Public Shared Sub TestDriveProperties_AppContainer()
+            Dim validDrive As IO.DriveInfo = FileSystem.Drives().Where(Function(d) d.DriveType = IO.DriveType.Fixed).First()
+            Dim isReady As Boolean = validDrive.IsReady
+            If PlatformDetection.IsInAppContainer Then
```

Comment:
break it into 2 tests with conditional fact

Reply:
I don't even know if there are AppContainers anymore. Is there a conditional fact for them?